content/blog/peace-of-mind-with-cloud-secret-providers/index.md

```diff
@@ -0,0 +1,669 @@
+		Principal Principal `json:"Principal"`
+		Action    []string  `json:"Action"`
+		Resource  string    `json:"Resource"`
+	}
+
+	type KeyPolicy struct {
+		Version    string      `json:"Version"`
+		ID         string      `json:"Id"`
+		Statements []Statement `json:"Statement"`
+	}
+
+	pulumi.Run(func(ctx *pulumi.Context) error {
+
+		config := config.New(ctx, "")
+
+		iamRole := config.Require("iamRole")
+		accountID := config.Require("accountID")
+
+		rootIamRole := fmt.Sprintf("arn:aws:iam::%s:root", accountID)
+
+		rawKeyPolicy := &KeyPolicy{
+			Version: "2012-10-17",
+			ID:      "policy",
+			Statements: []Statement{
+				{
+					Sid:    "AllowGetKeys",
+					Effect: "Allow",
+					Action: []string{
+						"kms:Describe*", "kms:Get*", "kms:List*",
+					},
+					Resource: "*",
+					Principal: Principal{
+						AWS: rootIamRole,
+					},
+				},
+				{
+					Sid:      "AllowIAMUserAccessKeys",
+					Effect:   "Allow",
+					Action:   []string{"kms:*"},
+					Resource: "*",
+					Principal: Principal{
+						AWS: iamRole,
+					},
+				},
+			},
+		}
+
+		keyPolicy, err := json.Marshal(rawKeyPolicy)
+
+		if err != nil {
+			panic("Error formatting keypolicy")
+		}
+
+		// Create an AWS KMS key
+		key, err := kms.NewKey(ctx, "stack-encryption-key", &kms.KeyArgs{
+			Description:          pulumi.String("KMS key for encrypting Pulumi secret values"),
+			DeletionWindowInDays: pulumi.Int(10),
+			Policy:               pulumi.String(string(keyPolicy)),
+		})
+		if err != nil {
+			return err
+		}
+
+		// Create an alias to the key
+		alias, err := kms.NewAlias(ctx, "alias/stack-encryption-key", &kms.AliasArgs{
+			TargetKeyId: key.KeyId,
+		})
+		if err != nil {
+			return err
+		}
+
+		// Export the name of the bucket
+		ctx.Export("keyArn", key.Arn)
+		ctx.Export("keyAlias", alias.Arn)
+		return nil
+	})
+}
+```
+{{% /choosable %}}
+
+{{% choosable language csharp %}}
+```csharp
+using Pulumi;
+using Kms = Pulumi.Aws.Kms;
+
+class KeyStack : Stack
+{
+    public KeyStack()
+    {
+        var config = new Config();
+        var iamRole = config.Require("iamRole");
+        var accountID = config.Require("accountID");
+
+        var keyPolicy = $@"{{
+            ""Version"": ""2012-10-17"",
+            ""Id"": ""policy"",
+            ""Statement"": [
+            {{
+                ""Sid"": ""AllowGetKeys"",
+                ""Effect"": ""Allow"",
+                ""Action"": [""kms:Describe*"", ""kms:Get*"", ""kms:List*""],
+                ""Principal"": {{
+                    ""AWS"": [""arn:aws:iam::{accountID}:root""]
+                }},
+                ""Resource"": ""*""
+            }},
+            {{
+                ""Sid"": ""AllowIAMUserAccessKeys"",
+                ""Effect"": ""Allow"",
+                ""Action"": [""kms:*""],
+                ""Principal"": {{
+                    ""AWS"": ""{iamRole}""
+                }},
+                ""Resource"": ""*""
+            }}
+            ]
+        }}";
```

Comment:
Any reason to be different than Go here?  Might make sense to be consistent.

Reply:
I think this is just because @mikhailshilkov wrote this for me (thanks @mikhailshilkov!)

Reply:
Should I convert to a dynamic object? It would probably look better indeed.